### PR TITLE
Delete more code used for supporting Python 2

### PR DESCRIPTION
**Pull request by ni-jfitzger**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- ~[ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~
- ~[ ] I've added tests applicable for this pull request~

### What does this Pull Request accomplish?

I noticed some more code that was an artifact of when we supported both Python 2 and Python 3. This deletes it.

### List issues fixed by this Pull Request below, if any.

None

### What testing has been done?

PR Checks

### CODE REVIEW

File: build/generate_template.py [hunk 1]
```diff
@@ -47,14 +46,9 @@ def generate_template(template_name, template_params, dest_file, in_zip_file=Fal
...
     logging.debug(rendered_template)
-    if sys.version_info.major < 3:
-        file_handle_public = codecs.open(dest_file, mode="w", encoding='utf-8')
-        file_handle_public.write(rendered_template)
-        file_handle_public.close()
-    else:
-        file_handle_public = open(dest_file, 'wb')
-        file_handle_public.write(bytes(rendered_template, "UTF-8"))
-        file_handle_public.close()
+    file_handle_public = open(dest_file, 'wb')
```
Comment (marcoskirsch) [inline build/generate_template.py:49]: use `with`?

File: tools/simple_mako.py [hunk 2]
```diff
@@ -49,14 +48,9 @@ def generate_template(template_name, template_params, dest_file):
...
     logging.debug(rendered_template)
-    if sys.version_info.major < 3:
-        file_handle_public = codecs.open(dest_file, mode="w", encoding='utf-8')
-        file_handle_public.write(rendered_template)
-        file_handle_public.close()
-    else:
-        file_handle_public = open(dest_file, 'wb')
-        file_handle_public.write(bytes(rendered_template, "UTF-8"))
-        file_handle_public.close()
+    file_handle_public = open(dest_file, 'wb')
```
Comment (marcoskirsch) [inline tools/simple_mako.py:51]: use `with` ?

**Review by marcoskirsch — CHANGES_REQUESTED** [review]

**Comment by marcoskirsch** [comment]
One day we should look at all our string formatting and replace with fstrings across the board.

**Comment by bkeryan** [comment]
> One day we should look at all our string formatting and replace with fstrings across the board.

[pyupgrade](https://pypi.org/project/pyupgrade/) can do this and other language upgrades automatically.

**Review by marcoskirsch — APPROVED** [review]

**Pull request merged**